x-pack/test/observability_functional/apps/observability/alerts/index.ts

```diff
@@ -6,32 +6,28 @@
  */
 
 import expect from '@kbn/expect';
-import querystring from 'querystring';
 import { FtrProviderContext } from '../../../ftr_provider_context';
 
-// Based on the x-pack/test/functional/es_archives/observability/alerts archive.
-const DATE_WITH_DATA = {
-  rangeFrom: '2021-08-31T13:36:22.109Z',
-  rangeTo: '2021-09-01T13:36:22.109Z',
-};
+async function asyncForEach<T>(array: T[], callback: (item: T, index: number) => void) {
```

Comment:
Is it important that these happen in sequence (versus in parallel with Promise.all)? 

Further, this is a big repo but I imagine this has been needed before. How common is it to extract common helpers like this on repo or "team" level?

Reply:
https://github.com/elastic/kibana/pull/111637

Reply:
We generally avoid extracting this type of stuff into repo level helpers. In the case that you want synchronous iteration I think that using a simple `for...of` loop makes the most sense.

```ts
for (const x of items) {
  await fn(x)
}
```